subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -601,6 +694,8 @@ static int cmd_rpl_clear(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+
+#if defined(CONFIG_BT_MESH_SHELL_PROV_CTX_INSTANCE)
```

Comment:
Shouldn't `struct bt_mesh_prov bt_mesh_shell_prov` be wrapped up in this macro either?

Reply:
Does it need to? Im not sure if extern variables will be compiled if they are not used anywhere 🤔

Reply:
Hm. Is it extern? https://github.com/Andrewpini/zephyr/blob/shell_prov_comp/subsys/bluetooth/mesh/shell/shell.c#L441-L460

Reply:
It is better to catch during the linking stage if you try to get access to a variable that is not used for provisioning. If it is kept as is you have a chance to use your own provisioning procedure but start beacon listening with a built-in one.

Reply:
Seems extern: https://github.com/zephyrproject-rtos/zephyr/blob/ffcbc391efab079383ed7dc3561ead06b0107485/include/zephyr/bluetooth/mesh/shell.h#L46-L47

Reply:
Yep, this allows it easy to build some samples like `mesh_shell` with `CONFIG_BT_MESH_SHELL_PROV_CTX_INSTANCE=n` and continue to use `bt_mesh_shell_prov`. This is an obvious misconfiguration. I'm pretty sur this should be fixed using condition compilation.